shared/pkg/utils/trivy/sourcetypes.go

```diff
@@ -36,3 +43,59 @@ func KubeclaritySourceToTrivySource(sourceType utils.SourceType) (artifact.Targe
 	}
 	return artifact.TargetKind("Unknown"), fmt.Errorf("unable to convert source type %v to trivy type", sourceType)
 }
+
+type CleanupFunc func(log *log.Entry)
+
+func UntarToTempDirectory(tar string) (string, CleanupFunc, error) {
+	tmpDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		return "", func(_ *log.Entry) {}, fmt.Errorf("unable to create temp directory: %w", err)
+	}
+	cleanup := func(log *log.Entry) {
+		err := os.RemoveAll(tmpDir)
+		if err != nil {
+			log.WithError(err).Errorf("unable to remove temp directory %s", tmpDir)
+		}
+	}
+
+	f, err := os.Open(tar)
+	if err != nil {
+		return "", cleanup, fmt.Errorf("unable to untar %s to temp directory: %w", tar, err)
+	}
+	err = stereoscopeFile.UntarToDirectory(f, tmpDir)
+	if err != nil {
+		return "", cleanup, fmt.Errorf("unable to untar %s to temp directory: %w", tar, err)
+	}
+	return tmpDir, cleanup, nil
+}
+
+func SetTrivyImageOptions(sourceType utils.SourceType, userInput string, trivyOptions trivyFlag.Options) (trivyFlag.Options, CleanupFunc, error) {
+	trivyOptions.ImageOptions = trivyFlag.ImageOptions{
+		ImageSources: types.AllImageSources,
```

Comment:
Do we need to set `ImageSources: types.AllImageSources` when it's an archive input type?

Reply:
good question, this is the default value for this field when using the Trivy CLI for any image input type (image, docker archive or OCI directory). Not sure what side effects if any it'll have if we unset it.

Reply:
I see, if that is the default in Trivy CLI so we can keep it also here, thanks.